test/widgets/emoji_reaction_test.dart

```diff
@@ -471,6 +469,87 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    group('handle view paddings', () {
+      const screenHeight = 400.0;
+
+      late double scrollViewTop;
+      late double scrollViewBottom;
+      final scrollViewFinder = findInPicker(find.bySubtype<ScrollView>());
+
+      late Widget firstOption;
+      late Widget lastOption;
+      double getScrollChildrenTop(WidgetTester tester) =>
+        tester.getTopLeft(find.byWidget(firstOption)).dy;
+      double getScrollChildrenBottom(WidgetTester tester) =>
+        tester.getBottomLeft(find.byWidget(lastOption)).dy;
+
+      Future<void> prepare(WidgetTester tester, {
+        required FakeViewPadding viewPadding,
+      }) async {
+        addTearDown(tester.view.reset);
+        tester.view.physicalSize = Size(640, screenHeight);
+        // This makes it easier to convert between device pixels used for
+        // [FakeViewPadding] and logical pixels used in tests.
+        tester.view.devicePixelRatio = 1.0;
+
+        tester.view.viewPadding = viewPadding;
+        tester.view.padding = viewPadding;
+
+        final message = eg.streamMessage();
+        await setupEmojiPicker(tester,
+          message: message, narrow: TopicNarrow.ofMessage(message));
+
+        scrollViewTop = tester.getTopLeft(scrollViewFinder).dy;
+        scrollViewBottom = tester.getBottomLeft(scrollViewFinder).dy;
+        // The scroll view should expand all the way to the bottom of the
+        // screen, even if there is device bottom padding.
+        check(scrollViewBottom).equals(screenHeight);
+
+        final options = tester.widgetList(find.byType(EmojiPickerListEntry));
+        firstOption = options.first;
+        lastOption = options.last;
+      }
+
+      testWidgets('no view padding', (tester) async {
+        await prepare(tester, viewPadding: FakeViewPadding.zero);
+
+        // The top of the list of children is padded by 8px;
+        // the bottom is out of view.
+        check(scrollViewTop).equals(getScrollChildrenTop(tester) - 8);
+        check(scrollViewBottom).isLessThan(getScrollChildrenBottom(tester));
```

Comment:
It looks like this assumes there's a very long list of children, but I'm not seeing that clearly in the setup code. Reading `setupEmojiPicker`, called by `prepare`, I think I see two items being set up?

```dart
      store.setServerEmojiData(ServerEmojiData(codeToNames: {
        '1f4a4': ['zzz', 'sleepy'], // (just 'zzz' in real data)
      }));
      await store.handleEvent(RealmEmojiUpdateEvent(id: 1, realmEmoji: {
        '1': eg.realmEmojiItem(emojiCode: '1', emojiName: 'buzzing'),
      }));
```

Reply:
The rest of them come from `EmojiStore.popularEmojiCandidates`.

The test controls the screen height so that the scroll view can be filled. I think we can add some additional checks in `prepare` to verify that's.